whipper/extern/asyncsub.py

```diff
@@ -149,28 +149,4 @@ def recv_some(p, t=.1, e=1, tr=5, stderr=0):
             y.append(r)
         else:
             time.sleep(max((x - time.time()) / tr, 0))
-    return ''.join(y)
-
-
-def send_all(p, data):
-    while data:
-        sent = p.send(data)
-        if sent is None:
-            raise Exception(message)
-        data = buffer(data, sent)
-
-
-if __name__ == '__main__':
-    if sys.platform == 'win32':
-        shell, commands, tail = ('cmd', ('dir /w', 'echo HELLO WORLD'), '\r\n')
-    else:
-        shell, commands, tail = ('sh', ('ls', 'echo HELLO WORLD'), '\n')
-
-    a = Popen(shell, stdin=PIPE, stdout=PIPE)
-    print(recv_some(a))
-    for cmd in commands:
-        send_all(a, cmd + tail)
-        print(recv_some(a))
-    send_all(a, 'exit' + tail)
-    print(recv_some(a, e=0))
-    a.wait()
+    return ''.join(x.decode() for x in y).encode()
```

Comment:
What’s the reason for no longer wrapping this in `if __name__ == '__main__'`?

Reply:
IIRC that was removed because there's no `buffer()` in Python 3 and it's unused code.